test/e2e/e2e_test.go

```diff
@@ -43,7 +44,9 @@ var _ = Describe("Workload cluster creation", func() {
 	})
 
 	AfterEach(func() {
-		dumpSpecResourcesAndCleanup(ctx, specName, bootstrapClusterProxy, artifactFolder, namespace, cluster, e2eConfig.GetIntervals, clusterName, clusterctlLogFolder, skipCleanup)
+		if false {
+			dumpSpecResourcesAndCleanup(ctx, specName, bootstrapClusterProxy, artifactFolder, namespace, cluster, e2eConfig.GetIntervals, clusterName, clusterctlLogFolder, skipCleanup)
```

Comment:
Actually this could be controlled by some environment variable, at the moment we don't even really use this, maybe in the future but currently we have a single "It" function so this runs only after everything has already finished but then in CI we would delete the VMs anyways so it would be nice if this would be configurable.

Anyways I think you should just remove the "if" after this goes from draft to regular PR and later we will add the configuration support here.